burr/core/application.py

```diff
@@ -980,10 +1000,19 @@ def sequence_id(self) -> Optional[int]:
 
         :return:
         """
-        return self._state.get(SEQUENCE_ID, 0)
+        return self._sequence_id
 
     def _increment_sequence_id(self):
-        self._state = self._state.update(**{SEQUENCE_ID: self.sequence_id + 1})
+        self._sequence_id = self._sequence_id + 1
```

Comment:
This is weird to me -- why are we keeping track of it in two places? We have `self._state` as a source of truth, we shouldn't have it as a field as well and keep it synced.